### PR TITLE
[5.7] PXC-3739: FLUSH TABLES FOR EXPORT lock not released when session ends

### PR DESCRIPTION
**Pull request by venkatesh-prasad-v**
This PR is for 5.7.
The 8.0 PR can be found [here](https://github.com/percona/percona-xtradb-cluster/pull/1551)

https://jira.percona.com/browse/PXC-3739
    
Problem:
FLUSH TABLE(S) FOR EXPORT command pauses the provider and is not resumed
when session gets disconnected (or killed).
    
In PXC context, pausing the provider refers to entering the galera local
local monitor, causing the local seqno to freeze and thereby blocking
other threads to commit. Similarily, resuming the cluster refers to
leaving the local monitor and allowing other waiting threads to proceed.
    
Solution:
Resume the provider if it was paused during FLUSH TABLE(S) FOR EXPORT
and when the session got disconnected (or killed).

Testing Done
---
galera suite: https://pxc.cd.percona.com/view/PXC%205.7/job/pxc-5.7-param/244/console

### CODE REVIEW

File: mysql-test/suite/galera/t/galera_flush_table_for_export.test [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+FLUSH TABLE t1 FOR EXPORT;
+--disconnect con2
+
+# Test that INSERT works.
+--connection node_1
+INSERT INTO t1 VALUES(1);
+
+# Test-2: Execute FLUSH TABLE for EXPORT and kill the session from another session.
+--connection con1
+--let $conn_id = `SELECT CONNECTION_ID();`
```
Comment (satya-bodapati) [inline mysql-test/suite/galera/t/galera_flush_table_for_export.test:29]: please check if normal FLUSH TABLE t1 is also affected?

**Review by satya-bodapati — CHANGES_REQUESTED** [review]
Looks good, minor comments

Also, Please elaborate a bit on the provider pausing in commit message

File: mysql-test/suite/galera/t/galera_flush_table_for_export.test (hunk 1 above)
Comment (venkatesh-prasad-v) [reply mysql-test/suite/galera/t/galera_flush_table_for_export.test:29]: Looks like the commands FLUSH TABLE t1 and LOCK TABLE t1 have no effect on the provider.

File: mysql-test/suite/galera/r/galera_flush_table_for_export_thread_pool.result [hunk 2]
```diff
@@ -0,0 +1,11 @@
+CREATE TABLE t1 (i INT PRIMARY KEY);
+FLUSH TABLE t1 FOR EXPORT;
+INSERT INTO t1 VALUES(1);
+FLUSH TABLE t1 FOR EXPORT;
+KILL 11;
```
Comment (satya-bodapati) [inline mysql-test/suite/galera/r/galera_flush_table_for_export_thread_pool.result]: we might have to mask connection id?

File: mysql-test/suite/galera/r/galera_flush_table_for_export.result [hunk 3]
```diff
@@ -0,0 +1,11 @@
+CREATE TABLE t1 (i INT PRIMARY KEY);
+FLUSH TABLE t1 FOR EXPORT;
+INSERT INTO t1 VALUES(1);
+FLUSH TABLE t1 FOR EXPORT;
+KILL 11;
```
Comment (satya-bodapati) [inline mysql-test/suite/galera/r/galera_flush_table_for_export.result]: masking of connection id required?

**Review by satya-bodapati — CHANGES_REQUESTED** [review]
one minor issue

File: mysql-test/suite/galera/r/galera_flush_table_for_export_thread_pool.result (hunk 2 above)
Comment (venkatesh-prasad-v) [reply mysql-test/suite/galera/r/galera_flush_table_for_export_thread_pool.result]: Fixed.

File: mysql-test/suite/galera/r/galera_flush_table_for_export.result (hunk 3 above)
Comment (venkatesh-prasad-v) [reply mysql-test/suite/galera/r/galera_flush_table_for_export.result]: Fixed

**Review by satya-bodapati — APPROVED** [review]
All good. approved

**Pull request merged**